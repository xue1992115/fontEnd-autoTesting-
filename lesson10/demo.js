export const timer = (callback) => {
  setTimeout(() => {
    callback();
  }, 8000);
};

export const timer2 = (callback) => {
  setTimeout(() => {
    callback();
    setTimeout(() => {
      callback();
    }, 3000);
  }, 3000);
};
