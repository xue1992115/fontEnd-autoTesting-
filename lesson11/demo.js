class Demo {
  a() {
    // ...异常复杂
  }
  b() {
    // ...异常复杂
  }
}
export default Demo;
