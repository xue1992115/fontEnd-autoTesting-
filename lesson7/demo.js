export const generateConfig = () => {
  return {
    server: "http://localhost",
    port: 8080,
    domain: "localhost",
    name: "test",
    time: "2023",
    data: new Date(),
  };
};
export const generateConfig2 = () => {
  return {
    server: "http://localhost",
    port: 8080,
    domain: "localhost",
    name: "test",
    time: "2023",
    data: new Date(),
  };
};

