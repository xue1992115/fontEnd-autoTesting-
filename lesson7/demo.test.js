import { generateConfig, generateConfig2 } from "./demo";
// 快照
test("测试 generateConfig函数", () => {
  expect(generateConfig()).toMatchSnapshot({
    data: expect.any(Date),
  });
});
// 快照 生成的快照不会新生成一个文件，而是添加到测试用例下方
// test("测试 generateConfig2函数", () => {
//   expect(generateConfig2()).toMatchInlineSnapshot({
//     data: expect.any(Date),
//   });
// });
