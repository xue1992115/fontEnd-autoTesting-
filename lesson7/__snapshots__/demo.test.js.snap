// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateConfig函数 1`] = `
Object {
  "data": Any<Date>,
  "domain": "localhost",
  "name": "test",
  "port": 8080,
  "server": "http://localhost",
  "time": "2023",
}
`;
